### ISSUE TITLE
ProxyRoute.GetConfigHash throws for empty Metadata

### ISSUE THREAD

**Issue by davidni** [microsoft/reverse-proxy#373]
### Describe the bug
A route with a non-null but empty Metadata causes ProxyRoute.GetConfigHash to throw. We need to make sure that a route that passes IConfigValidator validation shouldn't throw later on the stack.

```
System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.Aggregate[TSource](IEnumerable`1 source, Func`3 func)
   at Microsoft.ReverseProxy.Abstractions.ProxyRoute.GetConfigHash() in D:\github\microsoft-reverse-proxy\src\ReverseProxy\Abstractions\Routes\ProxyRoute.cs:line 126
   at Microsoft.ReverseProxy.Service.RuntimeRouteBuilder.Build(ProxyRoute source, ClusterInfo cluster, RouteInfo runtimeRoute) in D:\github\microsoft-reverse-proxy\src\ReverseProxy\Service\Config\RuntimeRouteBuilder.cs:line 56
   at Microsoft.ReverseProxy.Service.Management.ProxyConfigManager.<>c__DisplayClass24_2.<UpdateRuntimeRoutes>b__0(RouteInfo route) in D:\github\microsoft-reverse-proxy\src\ReverseProxy\Service\Management\ProxyConfigManager.cs:line 399
   at Microsoft.ReverseProxy.Service.Management.ItemManagerBase`1.GetOrCreateItem(String itemId, Action`1 setupAction) in D:\github\microsoft-reverse-proxy\src\ReverseProxy\Service\Management\ItemManagerBase.cs:line 48
   at Microsoft.ReverseProxy.Service.Management.ProxyConfigManager.UpdateRuntimeRoutes(IList`1 routes) in D:\github\microsoft-reverse-proxy\src\ReverseProxy\Service\Management\ProxyConfigManager.cs:line 380
   at Microsoft.ReverseProxy.Service.Management.ProxyConfigManager.<ApplyConfigAsync>d__19.MoveNext() in D:\github\microsoft-reverse-proxy\src\ReverseProxy\Service\Management\ProxyConfigManager.cs:line 147
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.ReverseProxy.Service.Management.ProxyConfigManager.<InitialLoadAsync>d__16.MoveNext() in D:\github\microsoft-reverse-proxy\src\ReverseProxy\Service\Management\ProxyConfigManager.cs:line 82
```

### Further technical details

- Latest master (commit `8841dac2a64b90a0eb6144f229885e832110bace`)
- Platform: all
- Bug is here:

https://github.com/microsoft/reverse-proxy/blob/8841dac2a64b90a0eb6144f229885e832110bace/src/ReverseProxy/Abstractions/RouteDiscovery/Contract/ProxyRoute.cs#L124-L128

**Comment by Tratcher** [comment]
Good thing there's already a plan (and PR) to replace this method.
https://github.com/microsoft/reverse-proxy/issues/359
https://github.com/microsoft/reverse-proxy/pull/366/files#diff-64c4bfdbb41e68af88306f432218d3e5R76

**Comment by karelz** [comment]
Will be fixed by @Kahbazi's PR #366
